src/MonoTorrent/MonoTorrent.Client.Messages/PeerMessage.cs

```diff
@@ -107,5 +107,10 @@ internal void Handle (TorrentManager manager, PeerId id)
         {
             manager.Mode.HandleMessage (id, this);
         }
+
+        public override string ToString()
+        {
+            return this.GetType().Name;
```

Comment:
The default ToString implementation should return the typename. https://docs.microsoft.com/en-us/dotnet/api/system.object.tostring?view=netframework-4.8#remarks . That should mean this change is unnecessary?